lib/apress/selenium_eti/spec/company_site/mini_eti/minieti_spec.rb

```diff
@@ -0,0 +1,200 @@
+      end
+
+      it 'введенные цены и дата окончания скидки отображаются' do
+        expect(@cs_eti_table_products.price(@product)).to include(@discount_price[:price], @discount_price[:new_price])
+      end
+    end
+
+    context 'когда заполняем наличие' do
+      before do
+        @product = @cs_eti_table_products.add_product(name: Faker::Number.number(5), exists: :available)
+      end
+
+      it 'для товара отобразится статус "В наличии"' do
+        expect(@cs_eti_table_products.exists(@product)).to match(/[Вв] наличии/)
+      end
+    end
+
+    context 'когда заполняем рубрику' do
+      before(:all) do
+        @product = @cs_eti_table_products.add_product(name: Faker::Number.number(5), rubric: CONFIG['eti']['rubric'])
+      end
+
+      it 'привязывается рубрика' do
+        expect(@cs_eti_table_products.rubric(@product)).to include CONFIG['eti']['rubric']
+      end
+
+      context 'когда отменяем действие' do
+        before(:all) { @cs_eti_action_panel.undo }
+
+        it 'рубрика исчезает' do
+          expect(@cs_eti_table_products.rubric(@product)).not_to include CONFIG['eti']['rubric']
+        end
+
+        context 'когда повторяем отмененное действие' do
+          before(:all) { @cs_eti_action_panel.redo }
+
+          it 'привязывается рубрика' do
+            expect(@cs_eti_table_products.rubric(@product)).to include CONFIG['eti']['rubric']
+          end
+        end
+      end
+    end
+  end
+
+  describe 'Пагинатор' do
+    context 'когда переходим на следующую страницу' do
+      before(:all) { @cs_mini_eti_pagination.next_page }
+
+      it 'открывается следующая страница' do
+        expect(no_page_errors?).to be true
+        expect(@cs_mini_eti_pagination.current_page.to_i).to eq(2)
+      end
+
+      context 'когда возвращаемся на предыдущую страницу' do
+        before(:all) { @cs_mini_eti_pagination.previous_page }
+
+        it 'открывается предыдущая страница' do
+          expect(no_page_errors?).to be true
+          expect(@cs_mini_eti_pagination.current_page.to_i).to eq(1)
+        end
+      end
+    end
+
+    context 'когда выбираем количество товаров на странице' do
+      before { @cs_mini_eti_pagination.per_page = 50 }
+
+      it 'количество товаров на странице равно выбранному значению' do
+        expect(@cs_eti_table_products.products_elements.size.between?(21, 50)).to be true
+      end
+    end
+  end
+
+  describe 'Удаление товара' do
+    before do
+      name = Faker::Number.number(5)
+      @product = @cs_eti_table_products.add_product(name: name)
+    end
+
+    it 'товар удаляется' do
+      expect { @cs_eti_table_products.delete_product(@product) } .to change { @product.visible? }.from(true).to(false)
+    end
+  end
+
+  describe 'Копирование товара' do
+    before do
+      @fields = {
+        name: Faker::Number.number(5),
+        price: {
+          type: :exact,
+          price: Faker::Number.number(5)
+        }
```

Comment:
Style/TrailingCommaInHashLiteral: Put a comma after the last item of a multiline hash.